### ISSUE TITLE
Try the first-order acoustic equation propagation

### ISSUE THREAD

**Issue by jedfang** [ar4/deepwave#59]
Hi Dr. Alan,
Thank you for your great contributions to seismic imaging and inversion.
I find that Deepwave does not seem to support propagation of first order acoustic equations. So I tried to write a forward kernel implementation using the GPU. I've done forward emulation in Pytorch with the kernel callback. However, when I tried to do FWI, I found that the gradient value of the training parameter was None when using loss.backward(). Is it necessary to write a reverse code for the network? When I read the description of Auto-gradient, it didn't seem necessary to do the reverse myself. Can you show me the way?
Thank you very much!

**Comment by jedfang** [comment]
Sure, I have sent the code via email. Thanks a lot for your kind guidance.
